packages/protocol/contracts/L1/TaikoConfig.sol

```diff
@@ -27,7 +27,8 @@ library TaikoConfig {
             blockMaxVerificationsPerTx: 10,
             // Set it to 6M, since its the upper limit of the Alpha-2
             // testnet's circuits.
-            blockMaxGasLimit: 6_000_000,
+            blockMaxGasUsed: 6_000_000,
+            blockMaxGasLimit: 12_000_000,
```

Comment:
Hmm shall we change it to `txMaxGasLimit`? looks like if there is a block gas limit, its still easy for malicious users to attack by simply increasing their `tx.gasLimit`?

Reply:
I agree, but I think the `txMaxGasLimit` should simply be `blockMaxGasUsed`, and then simply make all transactions with a gas limit higher than that invalid. 

Doing this we could avoid needing custom support for stopping the execution of a tx with a partial trace if we can prove blocks up till ` blockMaxGasUsed * 2`, which of course is not super efficient but may be worth it to reduce complexity, but I'm not really sure how big of a simplification it would actually be, I have to think about it some more.

And temporarily to be compatible with the current system, we can set `blockMaxGasLimit` to `blockMaxTransactions *  blockMaxGasUsed`.